x/evm/abci.go

```diff
@@ -65,25 +65,40 @@ func handleTokenSent(ctx sdk.Context, event types.Event, bk types.BaseKeeper, n
 }
 
 func handleContractCall(ctx sdk.Context, event types.Event, bk types.BaseKeeper, n types.Nexus, multisig types.MultisigKeeper) error {
-	e := event.GetEvent().(*types.Event_ContractCall).ContractCall
+	e := event.GetContractCall()
 	if e == nil {
 		panic(fmt.Errorf("event is nil"))
 	}
 
 	destinationChain := funcs.MustOk(n.GetChain(ctx, e.DestinationChain))
-	destinationCk := funcs.Must(bk.ForChain(ctx, destinationChain.Name))
+	switch destinationChain.Module {
+	case types.ModuleName:
+		handleContractCallToEVM(ctx, bk, multisig, n, destinationChain.Name, event)
+		return nil
+	default:
+		// set as general message in nexus, so the dest module can handle the message
+		return setGeneralMessageToNexus(ctx, n, event)
+	}
+}
+
+func handleContractCallToEVM(ctx sdk.Context, bk types.BaseKeeper, multisig types.MultisigKeeper, n types.Nexus, destinationChain nexus.ChainName, event types.Event) {
+	e := event.GetContractCall()
```

Comment:
need to validate that ContractAddress is actually an EVM address now that Validate doesn't do it, and this doesn't go through the nexus router yet

Reply:
added in validateEvent